### PR TITLE
Deprecate search

### PR DESCRIPTION
**Pull request by binaryoverload**

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java [hunk 1]
```diff
@@ -4,30 +4,15 @@
...
-            } else {
-                String term = "";
-                for (String s : args) {
-                    term += s + " ";
-                }
-                term = term.trim();
-                VideoThread.getSearchThread(term, channel, sender).start();
-            }
-        }
+        MessageUtils.sendErrorMessage("This command is deprecated!\nUse `_search` instead!", channel);
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java]: ok

Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java]: Oops
Will fix ;)

**Comment by binaryoverload** [comment]
@ArsenArsen Fixed ;)

File: src/main/java/stream/flarebot/flarebot/commands/music/ResumeCommand.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private PlayerManager musicManager;
+
+    public ResumeCommand(FlareBot bot) {
+        this.musicManager = bot.getMusicManager();
+    }
+
+    @Override
+    public void onCommand(User sender, TextChannel channel, Message message, String[] args, Member member) {
+        boolean playing = musicManager.getPlayer(channel.getGuild().getId()).getPlayingTrack() != null;
+        boolean paused = musicManager.getPlayer(channel.getGuild().getId()).getPaused();
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/ResumeCommand.java]: Inline these

File: src/main/java/stream/flarebot/flarebot/commands/music/ResumeCommand.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    private PlayerManager musicManager;
+
+    public ResumeCommand(FlareBot bot) {
+        this.musicManager = bot.getMusicManager();
+    }
+
+    @Override
+    public void onCommand(User sender, TextChannel channel, Message message, String[] args, Member member) {
+        boolean playing = musicManager.getPlayer(channel.getGuild().getId()).getPlayingTrack() != null;
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/ResumeCommand.java]: Inline these

File: src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java [hunk 4]
```diff
@@ -21,12 +22,25 @@ public PlayCommand(FlareBot bot) {
...
+                    term += s + " ";
+                }
+                term = term.trim();
+                VideoThread.getSearchThread(term, channel, sender).start();
+            }
         } else {
-            musicManager.getPlayer(channel.getGuild().getId()).play();
-            channel.sendMessage("Resuming...!").queue();
+            boolean playing = musicManager.getPlayer(channel.getGuild().getId()).getPlayingTrack() != null;
+            boolean paused = musicManager.getPlayer(channel.getGuild().getId()).getPaused();
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java]: Inline this

File: src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java [hunk 5]
```diff
@@ -21,12 +22,25 @@ public PlayCommand(FlareBot bot) {
...
+                for (String s : args) {
+                    term += s + " ";
+                }
+                term = term.trim();
+                VideoThread.getSearchThread(term, channel, sender).start();
+            }
         } else {
-            musicManager.getPlayer(channel.getGuild().getId()).play();
-            channel.sendMessage("Resuming...!").queue();
+            boolean playing = musicManager.getPlayer(channel.getGuild().getId()).getPlayingTrack() != null;
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java]: Inline this

**Comment by binaryoverload** [comment]
@ArsenArsen Inlining it would make the lines too long. Is there any reason why you want them inlined?

**Comment by ArsenArsen** [comment]
Indeed. Make a line break afterwards.

**Comment by binaryoverload** [comment]
@WalshyDev Yeah ok, will make it function as usual

**Comment by binaryoverload** [comment]
@ArsenArsen @WalshyDev Better?

File: src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java [hunk 6]
```diff
@@ -1,33 +1,30 @@
...
-                String term = "";
-                for (String s : args) {
-                    term += s + " ";
-                }
-                term = term.trim();
-                VideoThread.getSearchThread(term, channel, sender).start();
-            }
-        }
+        channel.sendMessage(new EmbedBuilder().setColor(Color.YELLOW).setDescription("This is deprecated! Please use _play instead!").build());
+        new PlayCommand(bot).onCommand(sender, channel, message, args, member);
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java]: No. Please. Just make a static method to process this.

**Comment by binaryoverload** [comment]
@ArsenArsen Happy?

File: src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java [hunk 7]
```diff
@@ -27,6 +36,14 @@ public void onCommand(User sender, TextChannel channel, Message message, String[
                 term = term.trim();
                 VideoThread.getSearchThread(term, channel, sender).start();
             }
+        } else {
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java]: Not yet. This was never the behaviour of `_search`

Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/music/SearchCommand.java]: Why are you resuming music in the search command??

**Comment by binaryoverload** [comment]
Fixed it

**Comment by binaryoverload** [comment]
xD

File: src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java [hunk 8]
```diff
@@ -21,12 +22,24 @@ public PlayCommand(FlareBot bot) {
     @Override
     public void onCommand(User sender, TextChannel channel, Message message, String[] args, Member member) {
         if (args.length > 0) {
-            MessageUtils
-                    .sendErrorMessage(MessageUtils.getEmbed().setDescription("To search for a song by term or URL do "
-                            + FlareBot.getPrefixes().get(channel.getGuild().getId()) + "search <term/URL>"), channel);
+            if (args[0].startsWith("http") || args[0].startsWith("www.")) {
+                VideoThread.getThread(args[0], channel, sender).start();
+            } else {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java]: Replace this with our new helper method FlareBot.getMessage

Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java]: @binaryoverload 
No idea if you saw this

Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java]: What so FlareBot.getMessage(args,0)

Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/music/PlayCommand.java]: Yeah

**Comment by binaryoverload** [comment]
@WalshyDev Saw what?

**Comment by WalshyDev** [comment]
The getMessage, GitHub was being weird.

**Comment by binaryoverload** [comment]
@WalshyDev Why use getMessage(args,0) over args[0] ?

**Comment by binaryoverload** [comment]
@WalshyDev Happy?

**Comment by binaryoverload** [comment]
YAY!

**Comment by binaryoverload** [comment]
Only took 8 commits!

**Pull request merged**